File: javascript/easy/minimumIndexSumOfTwoLists.js
/**
 * @param {string[]} list1
 * @param {string[]} list2
 * @return {string[]}
 */
var findRestaurant = function(list1, list2) {
    const map = new Map();
    let minSum = list1.length + list2.length, res = new Array();
    for(let i = 0; i < list1.length; i++) {
        map.set(list1[i], i);
    }
    
    for(let i = 0; i < list2.length; i++) {
        if(map.has(list2[i])) {
            minSum = Math.min(minSum, (map.get(list2[i]) + i));
        }
    }
    
    for(let i = 0; i < list2.length; i++) {
        if(map.has(list2[i]) && ((map.get(list2[i]) + i) === minSum)) {
            res.push(list2[i]);
        }
    }
    return res;
};

File: ruby/easy/minimum_index_sum_of_two_lists.rb
# @param {String[]} list1
# @param {String[]} list2
# @return {String[]}
def find_restaurant(list1, list2)
    map,res = {}, []
    min_index_sum = (list1.length + list2.length)
    list1.each_with_index do |l, i|
        map[l] = i
    end
    list2.each_with_index do |l, i|
        min_index_sum = [min_index_sum, (map[l] + i)].min if map[l]
    end
    
    list2.each_with_index do |l,i|
        res.push(l) if map[l] && ((map[l] + i) == min_index_sum)
    end
    return res
end
